traffic_ops/traffic_ops_golang/role/roles.go

```diff
@@ -273,3 +300,332 @@ RETURNING id, last_updated`
 func deleteQuery() string {
 	return `DELETE FROM role WHERE id = :id`
 }
+
+// Update will modify the role identified by the role name.
+func Update(w http.ResponseWriter, r *http.Request) {
+	var roleID int
+	var roleName string
+	var roleDesc string
+	var roleCapabilities []string
+	var roleV4 tc.RoleV4
+	var ok bool
+	var err error
+
+	inf, userErr, sysErr, errCode := api.NewInfo(r, nil, nil)
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+
+	tx := inf.Tx.Tx
+	if err := json.NewDecoder(r.Body).Decode(&roleV4); err != nil {
+		api.HandleErr(w, r, tx, http.StatusBadRequest, err, nil)
+		return
+	}
+	if err := roleV4.Validate(); err != nil {
+		api.HandleErr(w, r, tx, http.StatusBadRequest, err, nil)
+		return
+	}
+	missing := inf.User.MissingPermissions(roleV4.Permissions...)
+	if len(missing) != 0 {
+		api.HandleErr(w, r, tx, http.StatusForbidden, fmt.Errorf("cannot request more than assigned permissions, current user needs %s permissions", strings.Join(missing, ",")), nil)
+		return
+	}
+	roleDesc = roleV4.Description
+	roleCapabilities = roleV4.Permissions
+	if roleName, ok = inf.Params["name"]; !ok {
```

Comment:
if a string appears in the "required Parameters" argument of `api.NewInfo`, it'll do this check for you.

So instead of
```go
inf, userErr, sysErr, errCode := api.NewInfo(r, nil, nil)
```
doing
```go
inf, userErr, sysErr, errCode := api.NewInfo(r, []string{"name"}, nil)
```
will guarantee that `inf.Params` contains `"name"` as a key (or the proper error(s)/response code will be returned from `NewInfo`).